pkgs/tools/virtualization/guestfs-tools/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchurl
+, bison
+, cdrkit
+, cpio
+, flex
+, getopt
+, hivex
+, jansson
+, libguestfs
+, libvirt
+, libxml2
+, makeWrapper
+, ncurses
+, ocamlPackages
+, pcre2
+, perlPackages
+, pkg-config
+, qemu
+, xz
+}:
+
+stdenv.mkDerivation rec {
+  pname = "guestfs-tools";
+  version = "1.48.2";
+
+  src = fetchurl {
+    url = "https://download.libguestfs.org/${pname}/${lib.versions.majorMinor version}-stable/${pname}-${version}.tar.gz";
+    sha256 = "sha256-G9l5sG5g5kMlSXzg0GX8+Et7M9/k2dRLMBgsMI4MaxA=";
+  };
+
+  nativeBuildInputs = [
+    bison
+    cdrkit
+    cpio
+    flex
+    getopt
+    makeWrapper
+    pkg-config
+    qemu
+  ] ++
+  (with perlPackages; [
+    GetoptLong
+    libintl-perl
+    ModuleBuild
+    perl
+    Po4a
+  ]) ++
+  (with ocamlPackages; [
+    findlib
+    gettext-stub
+    ocaml
+    ocaml_gettext
+    ounit2
+  ]);
+
+  buildInputs = [
+    hivex
+    jansson
+    libguestfs
+    libvirt
+    libxml2
+    ncurses.dev
+    pcre2.dev
+    xz.dev
+  ];
+
+  enableParallelBuilding = true;
+
+  preBuild = ''
+    patchShebangs .
```

Comment:
This should be fine in postPatch and be a bit more precise than everything.

Reply:
I can move to postPatch, but it really needs to be everything that it is an executable (and this is a check that `patchShebangs` already do, so it doesn't make sense to create a pre-check just for it).

Reply:
You're right, its a bit all over the place https://cs.github.com/libguestfs/libguestfs?q=%23!%2Fbin%2Fbash&p=1